### ISSUE TITLE
why my cpu usage is such low when training

### ISSUE THREAD

**Issue by aitech17** [microsoft/LightGBM#6078]
## Description
why my cpu usage is such low when training? my cpu is 5950x

## Reproducible example
![5950](https://github.com/microsoft/LightGBM/assets/56558506/aae4aea8-0a48-47e5-91e7-534f8ad3dfb3)

**Comment by jameslamb** [comment]
Thanks for using LightGBM.

This image is not enough information for us to help much. You haven't even told us what programming language you are using, what operating system you're on, and what version of LightGBM you are using.

Please ensure you are passing an appropriate value to parameter `num_threads` and that OpenMP is installed on your system. See the discussion in https://github.com/microsoft/LightGBM/issues/4961#issuecomment-1016933389 for some help.

Otherwise, please provide a minimal, reproducible example ([docs link](https://stackoverflow.com/help/minimal-reproducible-example)) and we can try to investigate more.

**Comment by aitech17** [comment]
I use python package with pip install in windows, and I alse build lightgbm in wsl2, both have this problem. My params show below, and I use my own metric 
`params = {
        'objective': 'binary',
        'metric': "binary_logloss",
        'boosting': 'dart',
        'seed': CFG.seed,
        'num_leaves': 200,
        'learning_rate': 0.05,
        "subsample":0.8,
        "subsample_freq":1,
        "feature_fraction_bynode":0.7,
        'feature_fraction': 0.7,
        'bagging_freq': 10,
        'bagging_fraction': 0.80,
        'n_jobs': -1,
        'num_threads': 32,
        'lambda_l2': 2,
        'min_data_in_leaf': 2**11,
        }`

`def amex_metric(y_true, y_pred):
    labels = np.transpose(np.array([y_true, y_pred]))
    labels = labels[labels[:, 1].argsort()[::-1]]
    weights = np.where(labels[:,0]==0, 20, 1)
    cut_vals = labels[np.cumsum(weights) <= int(0.04 * np.sum(weights))]
    top_four = np.sum(cut_vals[:,0]) / np.sum(labels[:,0])
    gini = [0,0]
    for i in [1,0]:
        labels = np.transpose(np.array([y_true, y_pred]))
        labels = labels[labels[:, i].argsort()[::-1]]
        weight = np.where(labels[:,0]==0, 20, 1)
        weight_random = np.cumsum(weight / np.sum(weight))
        total_pos = np.sum(labels[:, 0] *  weight)
        cum_pos_found = np.cumsum(labels[:, 0] * weight)
        lorentz = cum_pos_found / total_pos
        gini[i] = np.sum((lorentz - weight_random) * weight)
    return 0.5 * (gini[1]/gini[0] + top_four)`

**Comment by jameslamb** [comment]
Thanks, but we need more details than that. For example... are you using `lightgbm.train()` or `lightgbm.LGBMClassifier.fit()`?

Do you see better CPU utilization if you use a built-in metric like `"auc"`? Maybe your custom metric is the bottleneck.

What version of LightGBM are you using?

----

I've reformatted your most recent post, to make the difference between code, output from code, and your own words clearer. See https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax to learn how I did that.
